source/GrowthbeatCore/GrowthbeatCore/GBCustomIntentHandler.m

```diff
@@ -0,0 +1,36 @@
+//
+//  GBCustomIntentHandler.m
+//  GrowthbeatCore
+//
+//  Created by TABATAKATSUTOSHI on 2015/07/16.
+//  Copyright (c) 2015年 SIROK, Inc. All rights reserved.
+//
+
+#import "GBCustomIntentHandler.h"
+#import "GrowthbeatCore.h"
+#import "GBLogger.h"
+
+@implementation GBCustomIntentHandler
+@synthesize block;
+
+- (BOOL) handleIntent:(GBIntent *)intent {
+    
+    if (intent.type != GBIntentTypeCustom) {
+        return NO;
+    }
+    if (self.block) {
+        self.block(intent);
+    } else {
+        [[[GrowthbeatCore sharedInstance] logger] error:@"error:GBCustomIntentHandler cannot handle intent. cause: block is nil"];
+    }
```

Comment:
When the intent is not processed successfully, return NO to pass the responsibility of the intent handling.

```objc
if (!self.block) {
    // TODO log here
    return NO;
}
```